packages/hardhat-core/test/internal/hardhat-network/provider/forked-provider.ts

```diff
@@ -539,6 +556,38 @@ describe("Forked provider", () => {
           });
         });
       });
+
+      describe("blocks timestamps", () => {
+        it("should use a timestamp relative to the forked block timestamp", async function () {
```

Comment:
I don't know if there is a better way of testing this. I'm using the first block of 2021-01-01, mining a new one, and checking that this new block has a timestamp on that day.